File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    answer = []
    temp = -1
    
    for i in arr:
        if answer[-1:] == [i]:
            continue
        answer.append(i)
        # if i == temp:
        #     continue
        # else:
        #     answer.append(i)
        #     temp = i
            
    return answer

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        Stack<Integer> stack = new Stack<>();
        
        for (int i = 0; i < arr.length; i++) {
            int now_num = arr[i];
            
            if (stack.size() != 0 && stack.peek() != now_num) {
                stack.push(now_num);
            } else if (stack.size() == 0) {
                stack.push(arr[i]);
            }
        }
        
        int[] result = new int[stack.size()];
        
        for (int i = stack.size() - 1; i >= 0; i--) {
            result[i] = stack.pop();
        }
        
        
        return result;
        
    }
}